lib/transaction.ex

```diff
@@ -225,4 +225,23 @@ defmodule Braintree.Transaction do
         {:error, Error.construct(%{"message" => "transaction id is invalid"})}
     end
   end
+
+  @doc """
+  Convert a map into a Transaction struct.
+  Add_ons are converted to a list of structs as well.
+
+  ## Example
+
+      transaction = Braintree.Transaction.construct(%{"subscription_id" => "subxid",
+                                                      "status" => "submitted_for_settlement"})
+  """
+  def construct(map_or_array) do
```

Comment:
I believe I'll need to restructure how `construct` gets used so that these overrides aren't necessary. In the meantime, can you please define this as two functions with guards, rather than as a single function with a case.